alonzo/formal-spec/txinfo.tex

```diff
@@ -3,5 +3,232 @@ \section{TxInfo Construction}
 
 This section specifies exactly what parts of the transaction and ledger
 state are used by the $\fun{txInfo}$ function to construct the
-$\TxInfo$ term that gets passed as an
-argument to the Plutus interpreter.
+$\TxInfo$ term that, together with the script purpose for which the script
+is being run, gets passed as a $\Data$ argument to the Plutus interpreter.
+
+The functions in Figure \ref{fig:txinfo-funcs} are needed to build a $\Data$ summary of a transaction.
+In this Section we use the notation $\type{P.PlutusType}$ for the Plutus type
+$\type{PlutusType}$ (specified in the Plutus library) to distinguish it from ledger types.
+The type $\type{P.ScriptContext}$ represents
+the script context. The function $\type{P.toData}$ converts a Plutus term
+into into a $\Data$ term.
+
+$\fun{txInfo}$ summarizes all the necessary transaction and chain state info
+    that needs to be passed to the script interpreter. Below, we specify how to
+    represent relevant transaction data as a term of a type defined in the Plutus
+    library. The $\fun{txInfo}$ function builds this representation.
+
+    The $\Language$ argument
+    is required because different languages have different expectations of the
+    format and contents of the $\TxInfo$ summary.
+
+    Note that $\fun{txInfo}$ has a $\UTxO$ argument. Even though the full ledger UTxO
+    is passed to it, we define this function in such a way that the only
+    entries in the ledger UTxO map that a Plutus script
+    actually sees via the argument $\fun{txInfo}$ builds are the ones corresponding to the transaction
+    inputs (ie. its realized inputs). This is done in order to maintain
+    deterministic evaluation. For details, see~\ref{sec:txinfo}.
```

Comment:
This link points to the current section :) Also, this is more about locality than about determinism.